### PR TITLE
Upgrade to Django 1.8

### PR DESCRIPTION
**Pull request by wedaly**
@symbolist please review when you have a moment.  Unfortunately, we won't be able to install this in edx-platform until we do the full Django 1.8 upgrade (because of the migrations in this app).

JIRA: [TNL-2829](https://openedx.atlassian.net/browse/TNL-2829)

### CODE REVIEW

**Comment by symbolist** [comment]
Looks good! :+1:

File: test.sh [hunk 1]
```diff
@@ -1,18 +1,4 @@
...
-IS_16=$(echo $DJ_VERSION | grep "1.6")
-
-# if django version is not 1.6 (non-0 exit) we have to pass different
-# app names to test runner
-if [ "$?" -ne "1" ]; then
-    app_names=( provider provider.oauth2 )
-else
-    app_names=( provider oauth2 )
-fi
-
```
Comment (nedbat) [inline test.sh:17]: Are we planning on providing these changes upstream?  Do we know what their Django version compatibility will be?  Maybe we don't want to remove all of this check.

**Comment by clintonb** [comment]
:+1:

File: test.sh (hunk 1 above)
Comment (justinabrahms) [inline test.sh:17]: The upstream library is not maintained.

**Pull request merged**